mobile-ui/src/main/java/org/sagebionetworks/research/mobile_ui/perform_task/PerformTaskFragment.java

```diff
@@ -219,7 +219,11 @@ void afterLastStep() {
         SharedPreferences sharedPreferences = getContext().getSharedPreferences(sharedPreferencesKey,
                 Context.MODE_PRIVATE);
         sharedPreferences.edit().putLong(LAST_RUN_KEY, Instant.now().toEpochMilli()).apply();
```

Comment:
This is outside the scope of the changes made here, but doesn't this line make it so the last run is always the time the user started this task? This seems like it would mean the whole logic around hiding instruction steps in walk and balance when the last run is less than a month ago won't work.

Reply:
This is in `afterLastStep()`, does that sound right?

Reply:
oh yeah I missed that, that does sound right to me.